model/task_queue_service.go

```diff
@@ -89,6 +89,9 @@ func (s *taskDispatchService) ensureQueue(distroID string) (CachedDispatcher, er
 	if err != nil {
 		return nil, errors.Wrapf(err, "database error for find() by distro id '%s'", distroID)
 	}
+	if d == nil {
+		return nil, errors.Errorf("distro '%s' not found", distroID)
+	}
 	// If there is a "distro": *basicCachedDispatcherImpl in the cachedDispatchers map, return that.
```

Comment:
This comment implies that it shouldn't be an error for distro not to exist, but rather that we'd just default to something else. Not super sure but might be worth keeping a non-pointer and just assigning *d to that if it's not nil